### DISCUSSION TITLE
Creating nodes from client's side

### DISCUSSION THREAD

**Discussion by EvgeBASS** [eclipse-milo/milo#910] [General]
Hello, and thanks you for all afforts. I am aspecially "young" in this theme, please said me how can i create new nodes from client's side?
I didn t fint appropriate example for this question( only editing).

**Comment by kevinherron** [comment]
There are no examples for this, but you would use the Node Management services provided by `OpcUaClient`:
- `OpcUaClient::addNodes`
- `OpcUaClient::addReferences`
- `OpcUaClient::deleteNodes`
- `OpcUaClient::deleteReferences`

You should read about these services in Part 4 of the OPC UA spec: https://reference.opcfoundation.org/v104/Core/docs/Part4/5.7.1/

**Reply by EvgeBASS** [reply]
Thanks for your responce! I started your ExampleServer class and tried to adding 1 new Variable node(Float) using WriteExample class as example. I also create all neccesery object to add node, but server response is: StatusCode{name=Bad_NotSupported, value=0x803D0000, quality=bad}.
Code in WriteExample:
        client.connect().get();
        ExpandedNodeId parentNodeId = new ExpandedNodeId(UShort.valueOf(2),null, "HelloWorld/ScalarTypes");
        NodeId referenceTypeId = new NodeId(Unsigned.ushort(0), Unsigned.uint(10));
        ExpandedNodeId requestedNewNodeId = new ExpandedNodeId(UShort.valueOf(2),null, "HelloWorld/ScalarTypes/TestFLOATNodee");
        QualifiedName browseName = new QualifiedName(2,"TestFLOATNodee");
        NodeClass nodeClass = NodeClass.Variable;
        ExtensionObject nodeAttributes = new ExtensionObject(new ByteString("4".getBytes()),referenceTypeId);
        ExpandedNodeId typeDefinition = new ExpandedNodeId(UShort.valueOf(2),null, "HelloWorld/ScalarTypes/Float");
        AddNodesItem addNodesItem = new 
  AddNodesItem(parentNodeId,referenceTypeId,requestedNewNodeId,browseName,nodeClass,nodeAttributes,typeDefinition);
        List<AddNodesItem> listNodes = new ArrayList<>();
        listNodes.add(addNodesItem);
        CompletableFuture<AddNodesResponse> responce = client.addNodes(listNodes);
        AddNodesResponse nodesresponse = responce.get();
        AddNodesResult[] results = nodesresponse.getResults();
        System.out.println(results[0].getStatusCode());
        future.complete(client);
Thanks for you patience... maybe i so bad...

**Reply by kevinherron** [reply]
The example server does not support these services. You’ll have to find one that does or implement them yourself.
